libsplinter/src/registry/diesel/operations/count_nodes.rs

```diff
@@ -32,32 +33,41 @@ where
     fn count_nodes(&self, predicates: &[MetadataPredicate]) -> Result<u32, RegistryError> {
         if predicates.is_empty() {
             // No predicates were specified, just count all nodes
-            splinter_nodes::table
+            let count = splinter_nodes::table
                 .count()
                 // Parse as an i64 here because Diesel knows how to convert a `BigInt` into an i64
                 .get_result::<i64>(self.conn)
-                .map(|count| count as u32)
                 .map_err(|err| {
                     RegistryError::general_error_with_source(
                         "Failed to count all nodes",
                         Box::new(err),
                     )
-                })
+                })?;
+
+            Ok(u32::try_from(count).map_err(|err| {
+                RegistryError::general_error_with_source("Failed to count all nodes", Box::new(err))
```

Comment:
As this is a pretty unlikely error, I think the message should be fairly specific. Something like "The number of nodes is larger than the max u32" (since count can't be negative), and we can omit the source, since I think the error type is uninteresting.

Reply:
updated

Reply:
updated